app/controllers/sessions_controller.rb

```diff
@@ -45,9 +45,13 @@ def authenticate_by_keycloak
     user = get_user(params['code'])
 
     if user.success?
+      HOMS.container[:cef_logger].log_user_event(:login, {id: user.value!.id, email: user.value!.email}, request.headers)
+
       sign_in(user.value!)
       redirect_to params.fetch(:redirect_to, '/')
     else
+      HOMS.container[:cef_logger].log_user_event(:failed_login, user, headers)
```

Comment:
you pass the `Failure` object as the second arg of `log_user_event`, it looks suspicious